core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -56,14 +58,30 @@ private[spark] class UnionPartition[T: ClassTag](
   }
 }
 
+object UnionRDD {
+  private lazy val partitionEvalThreadPool = new ForkJoinPool(8)
+}
+
 @DeveloperApi
 class UnionRDD[T: ClassTag](
     sc: SparkContext,
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // visible for testing
+  private[spark] val isPartitionEvalParallel: Boolean =
```

Comment:
I updated this to `shouldListPartitionsInParallel`. Starting with "read" makes it sound like it should do a read action. And using "list" matches the config property.